spec-tests/src/test/java/org/candlepin/spec/pools/OneSubPoolPerStackSpecTest.java

```diff
@@ -0,0 +1,805 @@
+    }
+
+    private ProductDTO createStackedDatacenterProduct(OwnerDTO owner, ProductDTO derivedProvidedProduct,
+        String stackId) {
+        return ownerProductApi.createProductByOwner(owner.getKey(), Products.random()
+            .attributes(List.of(
+                new AttributeDTO().name("virt_limit").value("unlimited"),
+                new AttributeDTO().name("stacking_id").value(stackId),
+                new AttributeDTO().name("multi-entitlement").value("yes"),
+                new AttributeDTO().name("sockets").value("2")
+            ))
+            .derivedProduct(createDerivedProduct(owner, createDerivedProduct(owner, derivedProvidedProduct)))
+            .providedProducts(Set.of(derivedProvidedProduct)));
+    }
+
+    private PoolDTO createStackedDatacenterProductPool(OwnerDTO owner, String stackId) {
+        return ownerApi.createPool(owner.getKey(), Pools.random(
+            createStackedDatacenterProduct(owner, ownerProductApi.createProductByOwner(owner.getKey(),
+            Products.randomEng()), stackId))
+            .startDate(OffsetDateTime.now().minusSeconds(5L))
+            .contractNumber(DATACENTER_POOL)
+            .accountNumber("")
+            .orderNumber(""));
+    }
+
+    private PoolDTO createStackedProductDiffIdPool(OwnerDTO owner, String stackId) {
+        return ownerApi.createPool(owner.getKey(), Pools.random(
+            createRegularStackedProduct(owner, stackId))
+            .quantity(2L)
+            .startDate(OffsetDateTime.now().minusDays(3L))
+            .endDate(OffsetDateTime.now().plusDays(6L))
+            .contractNumber(REGULAR_STACKED_WITH_DIFF_STACK_ID)
+            .accountNumber("")
+            .orderNumber(""));
+    }
+
+    private void createTargetPoolAndProducts(OwnerDTO owner, String stackingId) {
+        ContentDTO ownerContent1 = ownerContentApi.createContent(owner.getKey(), Content.random());
+        ContentDTO ownerContent2 = ownerContentApi.createContent(owner.getKey(), Content.random());
+        ProductDTO ownerDerivedEngProduct = ownerProductApi.createProductByOwner(
+            owner.getKey(), Products.random());
+        ownerProductApi.addBatchContent(owner.getKey(), ownerDerivedEngProduct.getId(),
+            Map.of(ownerContent1.getId(), true, ownerContent2.getId(), true));
+
+        ProductDTO ownerDerivedSku = ownerProductApi.createProductByOwner(owner.getKey(),
+            Products.random()
+            .multiplier(1L)
+            .attributes(List.of(new AttributeDTO().name("host_limited").value("true"))));
+
+        ProductDTO ownerSku = ownerProductApi.createProductByOwner(owner.getKey(),
+            Products.random()
+            .multiplier(1L)
+            .attributes(List.of(
+                new AttributeDTO().name("host_limited").value("true"),
+                new AttributeDTO().name("stacking_id").value(stackingId),
+                new AttributeDTO().name("virt_limit").value("unlimited"))));
```

Comment:
We should probably use the ProductAttributes enums here. That way if we were to have to change the key for whatever reason, then this test would receive that change as well:
`ProductAttributes.HostLimited.withValue("true");`

It looks like this will also need to be done in some other methods as well:
createStackedDatacenterProduct
createRegularStackedProduct
createVirtLimitProduct2
createDerivedProduct